src/libraries/Common/src/System/Collections/Generic/BitHelper.cs

```diff
@@ -19,19 +19,21 @@ internal BitHelper(Span<int> span, bool clear)
 
         internal void MarkBit(int bitPosition)
         {
-            int bitArrayIndex = bitPosition / IntSize;
-            if ((uint)bitArrayIndex < (uint)_span.Length)
+            uint bitArrayIndex = (uint)bitPosition / IntSize;
+            Span<int> span = _span;
+            if (bitArrayIndex < (uint)span.Length)
             {
-                _span[bitArrayIndex] |= (1 << (bitPosition % IntSize));
+                span[(int)bitArrayIndex] |= (1 << (int)((uint)bitPosition % IntSize));
             }
         }
 
         internal bool IsMarked(int bitPosition)
         {
-            int bitArrayIndex = bitPosition / IntSize;
+            uint bitArrayIndex = (uint)bitPosition / IntSize;
```

Comment:
Same for this one, its doing the same thing but via "more code" now. I'd expect the JIT was already hoisting the span since its a field of a struct.

Reply:
It is not the same thing. Unsigned division is a simple shift. Signed division is more complicated.

Reply:
> I'd also expect the JIT was already hoisting the span since its a field of a struct.

Unfortunately that's not the case*. Codegen was validated for this change.
\*  (at least when the PR was created, at the moment I don't have enough time to check again)

Reply:
The operation is "the same" even if the codegen is different. If we're doing this specifically to take advantage of the codegen, there should be a comment on it.

More generally, we also need some assert that the input is definitely positive -or- should be taking the input as `uint` and expecting the caller handle the validation/assertions. The current usages all look to be but its also always the output of `GetIndex` so without the corresponding outer checks, it'd be pretty easy for someone to accidentally pass in `-1` here from some refactoring.

If we're going to be making the method less readable, we should include a comment explaining why these tricks are being used (just as all the current cases have a small `// uint cast, per https://github.com/dotnet/runtime/issues/10596` or similar comment). Ideally the issue tracks ways to improve the JIT handling where we can't improve it ourselves via signature changes, locals, or other tweaks.